### ISSUE TITLE
Is future_apply not yet available?

### ISSUE THREAD

**Issue by seonghobae** [HenrikBengtsson/future.apply#6]
Hello, I love the ”future” API, and happy to release the ”future apply” package.

I want to use the futute_apply() function. But it seems not yet available. That's hard to implement in forthcoming? If that is so hard, how can I write alternative code using future() function? 

Sorry for my foolish and incompetence. 

Best,
Seongho

**Comment by HenrikBengtsson** [comment]
Yes, that's correct - there's no `future_apply()` yet (*).  The reason for it not been implemented yet is ... ... lack of time.  Also, contrary to many of the other `*apply()` functions, `apply()` is quite complex, e.g. argument `MARGIN` can be more than just a scalar.  This means quite a few package test cases need to be written to make sure it works correctly.

Depending on what `apply()` functionality you need, it may be more or less hard to do a workaround.  But if you're just after iteration over, say, columns in a matrix or a data.frame, you could try something like:

```r
X <- matrix(1:50, ncol = 5)

y0 <- apply(X, MARGIN = 2L, FUN = function(x, ...) {
  mean(x, ...)
})

y1 <- sapply(seq_len(ncol(X)), FUN = function(cc, ...) {
  mean(X[,cc], ...)
})

y2 <- future_sapply(seq_len(ncol(X)), FUN = function(cc, ...) {
  mean(X[,cc], ...)
})

stopifnot(identical(y1, y0), identical(y2, y0))
```


(*) I just realized that the package description mentions `apply()` - I've removed that in the develop branch.
